src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1477,13 +1477,14 @@ MonodroidRuntime::Java_mono_android_Runtime_initInternal (JNIEnv *env, jclass kl
 
 	if (androidSystem.is_mono_aot_enabled ()) {
 		MonoAotMode mode = androidSystem.get_mono_aot_mode ();
-		if (mode == MonoAotMode::MONO_AOT_MODE_LAST)
-			mode = MonoAotMode::MONO_AOT_MODE_NONE;
-
-		if (mode != MonoAotMode::MONO_AOT_MODE_NORMAL && mode != MonoAotMode::MONO_AOT_MODE_NONE) {
+		if (mode != MonoAotMode::MONO_AOT_MODE_LAST) {
 			log_info (LOG_DEFAULT, "Enabling AOT mode in Mono");
 			mono_jit_set_aot_mode (mode);
+		} else {
+			mono_jit_set_aot_mode (MonoAotMode::MONO_AOT_MODE_NONE);
 		}
+	} else {
+		mono_jit_set_aot_mode (MonoAotMode::MONO_AOT_MODE_NONE);
 	}
```

Comment:
I'd prefer if there was a single `mono_jit_set_aot_mode` call. Move the `mode` variable above the `if` on line 1478 and set it to `MonoAotMode::MONO_AOT_MODE_NONE` there. If AOT is enabled, proceed as we do now, but remove line 1482, then add a call to `mono_jit_set_aot_mode` after line 1488

Reply:
Ok, yeah this looks cleaner now, thanks.